snuba_sdk/conditions.py

```diff
@@ -144,3 +144,17 @@ class Or(BooleanCondition):
     conditions: Sequence[Union[BooleanCondition, Condition]] = field(
         default_factory=list
     )
+
+
+def get_top_level_conditions(
```

Comment:
Yep. It is flattening nested AND, since `(X AND Y) AND Z` is equivalent to `X AND Y AND Z`. I realized in Snuba we call it `get_first_level_and_conditions` so I'll change it though.